lupa/_lupa.pyx

```diff
@@ -1294,7 +1294,11 @@ cdef int raise_lua_error(LuaRuntime runtime, lua_State* L, int result) except -1
     elif result == lua.LUA_ERRMEM:
         raise MemoryError()
     else:
-        raise LuaError( build_lua_error_message(runtime, L, None, -1) )
+        error = py_from_lua(runtime, L, 1)
+        if isinstance(error, BaseException):
```

Comment:
Hmm. I'm not sure it's intentional that this retrieves a Python exception object. Couldn't such an exception also end up on the Lua stack for other reasons? Meaning, could this run into false positives?